### PR TITLE
Cleanup in 2M size distributions

### PR DESCRIPTION
**Pull request by trontrytel**
I noticed while doing integrals for the P3 scheme rates that the integral bound for cloud size distribution in SB2006 was returned in `mm`. I think it's ok to keep it in `m`, even though in some other places (aka effective radius and reflectivity calculations) we switch things to `ug` or `mm` for better accuracy. Keeping it in `m` will make it easier for P3 scheme computations.

### CODE REVIEW

**Comment by trontrytel** [comment]
Also, what I really care about in the tests are the integrals with bounds computed by the `get_size_distribution_bound` function. If you prefer, I can delete the other ones to simplify the tests

File: src/Microphysics2M.jl [hunk 1]
```diff
@@ -206,7 +206,7 @@ end
  - ρₐ - density of air
  - tolerance - tolerance for integration error
 
-    Returns D_max value such that (1 - tolerance) = 1/N * ∫ N'(D) dD from 0 to D_max.
+    Returns D_max value in meters such that (1 - tolerance) = 1/N * ∫ N'(D) dD from 0 to D_max.
```
Comment (sajjadazimi) [inline src/Microphysics2M.jl:209]: It would be good to say D_max value is in meters if inputs are in SI.

File: test/microphysics2M_tests.jl [hunk 2]
```diff
@@ -729,27 +734,32 @@ function test_microphysics2M(FT)
...
+        qₗ = FT(1e-3)      # 1 g/kg
 
         # air and liquid water densities in μg/m3
         ρₐ = FT(1.2)
         ρₗ = SB2006.pdf_r.ρw
 
-        # distribution parameters for cloud, units: [Bc] = 1/μg, [Ac] = 1/m3 1/μg3
+        # distribution parameters for cloud
+        # units: [Bc] = 1/μg, [Ac] = 1/m3 1/μg3
+        # units: [Ec] = 1/mm3, [Cc ] = (1/mm3)^4
```
Comment (sajjadazimi) [inline test/microphysics2M_tests.jl:762]: Based on line 774 (below), it seems like unit of Ec must be be 1/mm^\psi_c and unit of Cc also needs to be a function of \phi_c

**Review by sajjadazimi — APPROVED** [review]
Thank you for the additions. It looks good to me. I just left two small comments.

**Pull request merged**